src/esp32/board.rs

```diff
@@ -35,15 +39,119 @@ pub(crate) fn register_models(registry: &mut ComponentRegistry) {
     }
 }
 
+fn install_gpio_isr_service() -> anyhow::Result<()> {
+    static GPIO_ISR_SERVICE_INSTALLED: Lazy<Arc<OnceCell<()>>> =
+        Lazy::new(|| Arc::new(OnceCell::new()));
+    GPIO_ISR_SERVICE_INSTALLED.get_or_try_init(|| {
+        unsafe {
+            esp!(gpio_install_isr_service(ESP_INTR_FLAG_IRAM as i32))?;
+        };
+        Ok::<(), anyhow::Error>(())
+    })?;
+    Ok(())
+}
+
+/// GPIOPin is a wrapper for a pin on ESP32 as represented in esp-idf-hal
+/// and esp-idf-sys. This exists so that all micro-RDK drivers can interact
+/// with pins through the board instance and avoid conflicting uses of pins
+/// by multiple processes
+pub struct GPIOPin {
+    pin: i32,
+    driver: PinDriver<'static, AnyIOPin, InputOutput>,
+    pull: Pull,
+    interrupt_type: Option<InterruptType>,
+    event_count: Arc<AtomicU32>,
+}
+
+impl GPIOPin {
+    pub fn new(pin: i32, pull: Option<Pull>) -> anyhow::Result<Self> {
+        let mut driver = PinDriver::input_output(unsafe { AnyIOPin::new(pin) })?;
+        let pull = pull.unwrap_or(Pull::Up);
```

Comment:
this wouldn't get called when setting up I2C on the esp32?

Reply:
as mentioned below, I will change this so it does not pull the pin at all in the default behavior